modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+  $content_type = 'field_' . $conf['content_type'] . '_list_image';
+  $block = new stdClass();
+
+  // Prepare classes for widget container.
+  $classes = array();
+  $classes[] = 'ding_nodelist';
+  $classes[] = 'ding_nodelist-' . $conf['widget_type'];
+  $classes[] = $conf['unique_id'];
+
+  // Build title.
+  $title = NULL;
+  if (!empty($conf['override_title_text'])) {
+    $title = check_plain($conf['override_title_text']);
+  }
+
+  // Fetch content IDs.
+  $nodes = _ding_nodelist_get_custom_list_nodes($conf);
+
+  // Check if dynamic nodes must be loaded.
+  $custom_content = !empty($conf['selected_nodes']);
+  $dynamic_content = !empty($conf['taxonomy_filters']);
+  $only_custom_content = (($custom_content == TRUE) && ($dynamic_content == FALSE)) ? TRUE : FALSE;
+  if (!$only_custom_content) {
+    $nodes = array_merge($nodes, _ding_nodelist_get_dynamic_list_nodes($conf));
+  }
+
+  // Load items by their IDs.
+  $nodes = node_load_multiple($nodes);
+
+  // Remove elements if they are of type carousel, single item, vertical
+  // accordion or horizontal accordion and do not contain list image.
+  $widgets = array(
+    'single',
+    'carousel',
+    'horizontal_accordion',
+    'vertical_accordion',
+  );
+  if (in_array($conf['widget_type'], $widgets)) {
+    foreach ($nodes as $key => $node) {
+      if (count($node->{$content_type}) <= 0) {
+        unset($nodes[$key]);
+      }
+    }
+  }
+
+  if (count($nodes) > 0) {
+    // Apply sorting.
+    $nodes = _ding_nodelist_sort_nodes($nodes, $conf);
+    $nodes = _ding_nodelist_striptags($nodes);
+
+    $trimmed_widgets = _ding_nodelist_widgets_content_trim();
+    if (in_array($conf['widget_type'], $trimmed_widgets)) {
+      $nodes = _ding_nodelist_trim_body($nodes, $conf);
+    }
+  }
+
+  // Init JS for widgets.
+  _ding_nodelist_add_js($conf['widget_type'], $conf['autoscroll_delay'], $conf['unique_id']);
+  $image_style = _ding_nodelist_init_image_style($conf['widget_type']);
+  $theme_func = '_ding_nodelist_widget_' . $conf['widget_type'];
+  $block->content = theme($theme_func, array(
+    'items' => $nodes,
+    'template' => $conf['item_template'],
+    'links' => $conf['more_links'],
+    'conf' => array(
+      'title' => $title,
+      'classes' => implode(' ', $classes),
+      'image_style' => $image_style,
+      'sorting' => $conf['sort_field'],
+    ),
+  ));
+
+  return $block;
+}
+
+/**
+ * Add required JS libraries and initialization scripts to run widgets.
+ *
+ * @param string $widget
+ *   Widget type.
+ * @param int $autoscroll_delay
+ *   Animated widgets step delay.
+ * @param int $unique_id
+ *   Widget identifier.
+ */
+function _ding_nodelist_add_js($widget, $autoscroll_delay = NULL, $unique_id = NULL) {
+  $mod_path = drupal_get_path('module', 'ding_nodelist');
+  // Adding required JS libs.
+  switch ($widget) {
+    case NL_WIDGET_CAROUSEL:
+      drupal_add_js($mod_path . '/js/jquery.carouFredSel-5.5.0-packed.js', 'file');
```

Comment:
There are some external javascript libraries defined here which are also included in the PR. They should be included using Drush Make and Libraries.

Additionally `jquery.carouFredSel` is yet another carousel script. We are generally trying to use Slick (See #1672 + #1770). Can we use this instead?